tests/integration/base_block_test.go

```diff
@@ -111,10 +110,30 @@ func runBlockE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.
 	logger.Infof("Block Storage unmounted successfully")
 
 	logger.Infof("step 10: Deleting block storage")
-	_, dbErr := installer.BlockResourceOperation(k8sh, installer.GetBlockPoolStorageClassAndPvcDef(namespace, poolName, storageClassName, reclaimPolicy, blockName, "ReadWriteOnce"), "delete")
+	pvName, err := k8sh.GetPVCVolumeName(defaultNamespace, blockName)
+	pv, _ := k8sh.GetPV(pvName)
+	require.Nil(s.T(), err)
+	_, dbErr := installer.BlockResourceOperation(k8sh, installer.GetBlockPvcDef(blockName, storageClassName, "ReadWriteOnce"), "delete")
 	require.Nil(s.T(), dbErr)
-	require.True(s.T(), retryBlockImageCountCheck(helper, len(initBlockImages), 0, namespace), "Make sure a block is deleted")
-	logger.Infof("Block Storage deleted successfully")
+	if reclaimPolicy == "Delete" {
+		require.Equal(s.T(), fmt.Sprintf("%s", (*pv).Spec.PersistentVolumeReclaimPolicy), "Delete")
+		require.True(s.T(), retryPVCheck(k8sh, pvName, false, ""))
+		logger.Infof("PV: %s deleted successfully", pvName)
+		require.True(s.T(), retryBlockImageCountCheck(helper, len(initBlockImages), 0, namespace), "Make sure a block is deleted")
+		logger.Infof("Block Storage deleted successfully")
+	} else if reclaimPolicy == "Retain" {
+		require.Equal(s.T(), fmt.Sprintf("%s", (*pv).Spec.PersistentVolumeReclaimPolicy), "Retain")
+		require.True(s.T(), retryPVCheck(k8sh, pvName, true, "Released"))
+		time.Sleep(10 * time.Second)
```

Comment:
Is there any event we could look for, or at least cut down on the sleep time?

Reply:
@travisn Do we have some test util to catch events?

Reply:
@travisn I have seen that it usually takes some time for the `PV` to get deleted.